### PR TITLE
add maintenance mode

### PR DESCRIPTION
**Pull request by xcaptain**
if  a device is in maintenance mode, its credit won't change

### CODE REVIEW

**Comment by xcaptain** [comment]
Ready to review, please take a look

File: pallets/credit/src/lib.rs [hunk 1]
```diff
@@ -1331,6 +1370,11 @@ pub mod pallet {
 
         /// update credit score by traffic
         fn update_credit_by_traffic(server_id: T::AccountId) {
+            let addrs = MaintainDevices::<T>::get();
```
Comment (jerry-yu) [inline pallets/credit/src/lib.rs:1373]: maybe not need add this judge.  
If device sending back and admin do not remove device account from list in time

Comment (xcaptain) [reply pallets/credit/src/lib.rs:1373]: Good point, this check is not very important, will remove later.

**Review by jerry-yu — APPROVED** [review]

**Comment by xcaptain** [comment]
integration test failed, still trying to figure out why

**Comment by xcaptain** [comment]
@AmadeusGB Added a new privilege called `DeviceAdmin`, please take a look

**Comment by AmadeusGB** [comment]
@xcaptain LGTM

**Review by AmadeusGB — APPROVED** [review]

**Pull request merged**